### PR TITLE
Fix wrapper arguments escaping

### PR DESCRIPTION
**Pull request by ncfavier**
Some packages invoke `makeWrapper` like this:

```
wrapProgram foo --prefix PATH : ${bins}
```

The lack of escaping is a problem in cases where `bins` might be an empty string. The problem is becoming more apparent with https://github.com/NixOS/nixpkgs/pull/164163 merged, because `makeWrapper` used to silently do nothing if the third argument to `--prefix` is missing, while `makeBinaryWrapper` fails.

This PR fixes all instances of this pattern I could find where `bins` has a chance of being empty (hence leaving alone statically okay cases like `--prefix PATH : ${lib.makeBinPath [ hello ]}`).

ZHF https://github.com/NixOS/nixpkgs/issues/172160

### CODE REVIEW

**Comment by ncfavier** [comment]
@ofborg build image_optim megapixels chatty tremc jameica soapysdr qucs-s vdr podman i3blocks-gaps hunspell nuspell hci hercules-ci-agent sqitchMysql navidrome pufferpanel ibus-engines.typing-booster

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 171836` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
31 packages built:
</summary>
<ul>
<li>chatty</li>
<li>cubicsdr</li>
<li>gnuradio3_8Packages.ais</li>
<li>gnuradio3_8Packages.osmosdr</li>
<li>gqrx</li>
<li>gqrx-gr-audio</li>
<li>gqrx-portaudio</li>
<li>hci</li>
<li>hercules-ci-agent</li>
<li>i3blocks-gaps</li>
<li>ibus-engines.typing-booster</li>
<li>image_optim</li>
<li>jameica</li>
<li>megapixels</li>
<li>navidrome</li>
<li>openwebrx</li>
<li>podman</li>
<li>pothos</li>
<li>pufferpanel</li>
<li>python310Packages.soapysdr-with-plugins</li>
<li>python39Packages.soapysdr-with-plugins</li>
<li>qradiolink</li>
<li>qucs-s</li>
<li>rtl_433</li>
<li>sdrangel</li>
<li>soapysdr-with-plugins</li>
<li>sqitchMysql</li>
<li>sqitchPg</li>
<li>tremc</li>
<li>welle-io</li>
<li>wrapVdr</li>
</ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/applications/window-managers/i3/blocks-gaps.nix [hunk 1]
```diff
@@ -29,16 +29,26 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [ makeWrapper ];
 
   postFixup = ''
+    ${optionalString (elem "bandwidth" scripts) ''
```
Comment (dotlambda) [inline pkgs/applications/window-managers/i3/blocks-gaps.nix:32]:
```suggestion
  postFixup = optionalString (elem "bandwidth" scripts) ''
```

File: pkgs/applications/window-managers/i3/blocks-gaps.nix [hunk 2]
```diff
@@ -29,16 +29,26 @@ stdenv.mkDerivation rec {
...
     wrapProgram $out/libexec/i3blocks/bandwidth \
-      --prefix PATH : ${makeBinPath (optional (elem "bandwidth" scripts) iproute2)}
+      --prefix PATH : ${makeBinPath [ iproute2 ]}
+    ''}
+    ${optionalString (elem "battery" scripts) ''
```
Comment (dotlambda) [inline pkgs/applications/window-managers/i3/blocks-gaps.nix:36]:
```suggestion
    '' + optionalString (elem "battery" scripts) ''
```

File: pkgs/servers/misc/navidrome/default.nix [hunk 3]
```diff
@@ -33,8 +33,10 @@ stdenv.mkDerivation rec {
   '';
 
   postFixup = ''
+    ${optionalString ffmpegSupport ''
```
Comment (dotlambda) [inline pkgs/servers/misc/navidrome/default.nix:36]:
```suggestion
  postFixup = optionalString ffmpegSupport ''
```

File: pkgs/servers/misc/navidrome/default.nix [hunk 4]
```diff
@@ -33,8 +33,10 @@ stdenv.mkDerivation rec {
...
     wrapProgram $out/bin/navidrome \
-      --prefix PATH : ${makeBinPath (optional ffmpegSupport ffmpeg)}
+      --prefix PATH : ${makeBinPath [ ffmpeg ]}
+    ''}
```
Comment (dotlambda) [inline pkgs/servers/misc/navidrome/default.nix:39]:
```suggestion
```

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 171836` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>29 packages built:</summary>
  <ul>
    <li>chatty</li>
    <li>cubicsdr</li>
    <li>gnuradio3_8Packages.ais</li>
    <li>gnuradio3_8Packages.osmosdr</li>
    <li>gqrx</li>
    <li>gqrx-gr-audio</li>
    <li>gqrx-portaudio</li>
    <li>hci</li>
    <li>hercules-ci-agent</li>
    <li>ibus-engines.typing-booster</li>
    <li>image_optim</li>
    <li>jameica</li>
    <li>megapixels</li>
    <li>openwebrx</li>
    <li>podman</li>
    <li>pothos</li>
    <li>pufferpanel</li>
    <li>python310Packages.soapysdr-with-plugins</li>
    <li>python39Packages.soapysdr-with-plugins</li>
    <li>qradiolink</li>
    <li>qucs-s</li>
    <li>rtl_433</li>
    <li>sdrangel</li>
    <li>soapysdr-with-plugins</li>
    <li>sqitchMysql</li>
    <li>sqitchPg</li>
    <li>tremc</li>
    <li>welle-io</li>
    <li>wrapVdr</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

**Pull request merged**